src/app/user/user-details/user-details.component.ts

```diff
@@ -0,0 +1,112 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Observable, of, Subject } from 'rxjs';
+import { User } from '../../shared/models/user.model';
+import { UserService } from '../../shared/services/user.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { takeUntil, catchError } from 'rxjs';
+import { ParamMap } from '@angular/router';
+import { AsyncPipe } from '@angular/common';
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+import { MatTooltipModule } from '@angular/material/tooltip';
+import {MatCheckboxModule} from '@angular/material/checkbox';
+import { ReactiveFormsModule, FormControl, FormGroup } from '@angular/forms';
+
+@Component({
+  selector: 'user-details',
+  standalone: true,
+  imports: [AsyncPipe, MatIconModule, MatButtonModule, MatTooltipModule, MatCheckboxModule, ReactiveFormsModule],
+  templateUrl: './user-details.component.html',
+  styleUrl: './user-details.component.scss'
+})
+export class UserDetailsComponent implements OnInit, OnDestroy {
+  user$: Observable<User> | null = null;
+  private destroy$: Subject<void> = new Subject<void>();
+  editable: boolean = false;
+
+  roleForm = new FormGroup({
+    user: new FormControl({ value: false, disabled: true }),
+    manager: new FormControl(false)
+  });
+
+  constructor(private userService: UserService, private route: ActivatedRoute, private router: Router) {}
+
+  ngOnInit(): void {
+    this.route.paramMap
+    .pipe(takeUntil(this.destroy$))
+    .subscribe((params : ParamMap) => {
+      let id = params.get('id');
+
+      if (id) {
```

Comment:
you can improve this method by directly assigning to this.user$ by utilizing rjxs operators map and switchMap, first you would map the params to only return the id and then the next operator switchMap would get the id, which you can use to return the userDetails with this.userService.getUser()

Reply:
you wouldn't need to subscribe either because you would be doing that with the async pipe in the template!